## Binomial  Distribution
<h1>Assumptions of Binomial Distribution </h1>

* **The experiment involves n identical trials.**
* **Each trial has only two possible outcomes denoted as success or as failure.**
*  **Each trial is independent of the previous trials.**
* **The terms p and q remain constant throughout the experiment, where the term p
is the probability of getting a success on any one trial and the term q = (1 - p) is
the probability of getting a failure on any one trial.**

<h2>Formula for binomail distribution</h2>
$$P (x) = nCx\times p^x \times q^{n-x} =\frac {n!}{x!(n - x)!)} \times p^x q^{n-x} $$

where
* n = the number of trials (or the number being sampled)
* x = the number of successes desired
* p = the probability of getting a success in one trial
* q = 1 - p = the probability of getting a failure in one trial

 **$$\mu = np$$**
 **$$\sigma^2 =  npq$$**

In [ ]:
import numpy as np 
import scipy as sc
from scipy.stats import binom

**A Gallup survey found that 65% of all financial consumers were very satisfied with
their primary financial institution. Suppose that 25 financial consumers are sampled
and if the Gallup survey result still holds true today, what is the probability that
exactly 19 are very satisfied with their primary financial institution?**

n=25

x=19

p=0.65

q=0.35

In [ ]:
print(binom.pmf(k=19,n=25,p=0.65))

0.09077799859322791


probability that exactly 19 are very satisfied with their primary financial institution is 9.08 %

**According to the U.S. Census Bureau, approximately 6% of all workers in Jackson,Mississippi, are unemployed. In conducting a random telephone survey in Jackson, what is the probability of getting two or fewer unemployed workers in a sample of 20?** 

n=20

x<=2

p=0.06

q=0.94

In [ ]:
binom.cdf(2,20,0.06)

0.8850275957378545

**If n = 10 and p = .60, find P(x>=7).**

In [ ]:
prob=binom.pmf(k=7,n=10,p=0.60)+binom.pmf(k=8,n=10,p=0.60)+binom.pmf(k=9,n=10,p=0.60)+binom.pmf(k=10,n=10,p=0.60)
print(round(prob,3))

0.382


In [ ]:
prob=binom.cdf(6,10,0.60)
print(1-prob)

0.3822806016000001


**What is the first big change that American drivers made due to higher gas prices?
According to an Access America survey, 30% said that it was cutting recreational
driving. However, 27% said that it was consolidating or reducing errands. If these
figures are true for all American drivers, and if 20 such drivers are randomly sampled
and asked what is the first big change they made due to higher gas prices,**

**a. What is the probability that exactly 8 said that it was consolidating or reducing errands?**

**b. What is the probability that none of them said that it was cutting recreational
driving?**

**c. What is the probability that more than 7 said that it was cutting recreational
driving?**

In [ ]:
#a. What is the probability that exactly 8 said that it was consolidating or reducing errands?
print(binom.pmf(k=8,n=20,p=0.27))

0.08148010201492877


In [ ]:
#b. What is the probability that none of them said that it was cutting recreational driving?
print(round(binom.pmf(k=0,n=20,p=0.30),5))

0.0008


In [ ]:
#c. What is the probability that more than 7 said that it was cutting recreational driving?
prob=1-binom.cdf(7,20,0.30)
print(round(prob,3))

0.228
